uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js

```diff
@@ -574,15 +627,42 @@ define([
       vm.pagination.pageChanged();
     }
 
+    /**
+     * Performs a set of manipulations to prepare all data for calculation and
+     * then triggers the calculation itself if it can be calculated
+     *
+     * @return {Promise}
+     */
+    function attemptCalculateBalanceChange () {
+      updateOpeningBalance();
+      setHoursDeductionsToRequest();
+      setMinMaxDatesToUI();
```

Comment:
This looks very out of place. We are about to calculate the balance change, why do we bother with setting the min max dates in the UI? This needs to be moved somewhere else where it makes more sense

Reply:
Because we should only set min/max dates when we are sure in the absence period and we cannot be sure until attemptCalculateBalanceChange is called.

Reply:
to be discussed